Automattic/jetpack
# 3rd Party: improve loading

## Proposed changes:

* Updating the loading system to only include things that need to be loaded instead of loading everything with an over-reliance on hooks.

### Other information:


## Jetpack product discussion


p1HpG7-lOi-p2

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






* Needs performance profiling.
* bbPress: sharing links should appear under bbPress items (discussion threads, etc)
* Bitly: ensure that Bitly OG tags are removed, but I'm pretty sure that plugin is no longer available. 
* Creative Mail, Jetpack Backup, Jetpack Boost: ensure that JITMs to install these plugins work.
* On WoA, need to check it out on a Atomic dev site where we can edit the `jetpack` plugin. (`jetpack-dev` still should return as as a development site).
* ...
